### PR TITLE
Add compat data for api.TextDecoder

### PR DESCRIPTION
**Pull request by luciancic**
[TextDecoder](https://developer.mozilla.org/docs/Web/API/TextDecoder)

### CODE REVIEW

**Comment by connorshea** [comment]
@lucian95 is this a duplicate of #1919? (technically that's a duplicate of this, but the other one has my review on it so *shrug*)

**Comment by lucian95** [comment]
@connorshea  #1919 is Text**Encoder**, this one is Text**Decoder**

File: api/TextDecoder.json [hunk 1]
```diff
@@ -208,50 +449,113 @@
...
           "status": {
-            "experimental": false,
+            "experimental": true,
             "standard_track": true,
             "deprecated": false
           }
+        },
+        "worker_support": {
+          "__compat": {
+            "description": "Available in Web Workers",
```
Comment (connorshea) [inline api/TextDecoder.json]: The description should be "Available in workers", see https://github.com/mdn/browser-compat-data/blob/03dbe3ccbc283469e18fa2ebdaa647e3ee752a42/schemas/compat-data-schema.md#api-specific-subfeatures

File: api/TextDecoder.json [hunk 2]
```diff
@@ -88,17 +100,246 @@
...
+          },
+          "status": {
+            "experimental": true,
             "standard_track": true,
             "deprecated": false
           }
+        },
+        "worker_support": {
+          "__compat": {
+            "description": "Available in Web Workers",
```
Comment (connorshea) [inline api/TextDecoder.json]: The description should be "Available in workers"

File: api/TextDecoder.json [hunk 3]
```diff
@@ -88,17 +100,246 @@
...
           "status": {
-            "experimental": false,
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": false
+          }
+        },
+        "worker_support": {
+          "__compat": {
+            "description": "Available in Web Workers",
```
Comment (connorshea) [inline api/TextDecoder.json]: The description should be "Available in workers"

File: api/TextDecoder.json [hunk 4]
```diff
@@ -5,63 +5,75 @@
...
+          "experimental": true,
           "standard_track": true,
           "deprecated": false
         }
       },
-      "encoding": {
+      "worker_support": {
         "__compat": {
-          "mdn_url": "https://developer.mozilla.org/docs/Web/API/TextDecoder/encoding",
+          "description": "Available in Web Workers",
```
Comment (connorshea) [inline api/TextDecoder.json]: The description should be "Available in workers"

File: api/TextDecoder.json [hunk 5]
```diff
@@ -5,63 +5,75 @@
         "mdn_url": "https://developer.mozilla.org/docs/Web/API/TextDecoder",
         "support": {
           "webview_android": {
```
Comment (connorshea) [inline api/TextDecoder.json]: Ideally `webview_android` would be moved to the bottom so it's alphabetized but that's not a deal breaker imo

File: api/TextDecoder.json [hunk 6]
```diff
@@ -5,63 +5,75 @@
...
-          },
-          "firefox_android": {
-            "version_added": null
-          },
+          "firefox": [
+            {
+              "version_added": "19"
+            },
+            {
+              "version_added": "18",
```
Comment (connorshea) [inline api/TextDecoder.json]: This should probably be marked as a partial implementation?

**Pull request merged**